model/repository/MonitoringRepository.php

```diff
@@ -460,23 +460,37 @@ protected function prepareOrderStmt($order): string
     {
         $order = explode(',', $order);
         $result = [];
-        $primaryTableColumns = $this->getPrimaryColumns();
-        foreach ($order as $ruleNum => $orderRule) {
-            preg_match('/([a-zA-Z_][a-zA-Z0-9_]*)\s?(asc|desc)?\s?(string|numeric)?/i', $orderRule, $ruleParts);
+        foreach ($order as $orderRule) {
+            $result[] = $this->buildSingleOrderRule($orderRule);
+        }
 
-            if (!in_array($ruleParts[1], $primaryTableColumns)) {
-                $colName = $ruleParts[1];
-                $this->queryParams[] = $colName;
+        return implode(', ', $result);
+    }
+
+    private function buildSingleOrderRule(string $orderRule): string
+    {
+        preg_match(
+            '/([a-zA-Z_][a-zA-Z0-9_]*)\s?(asc|desc)?\s?(string|numeric)?/i',
+            $orderRule,
+            $ruleParts
+        );
+
+        if (!in_array($ruleParts[1], $this->getPrimaryColumns(), true)) {
+            $colName = $ruleParts[1];
+            if (in_array($this->getPlatformName(), ['mysql', 'sqlite'])) {
+                $colName = sprintf('JSON_EXTRACT(t.%s, \'$.%s\')', self::COLUMN_EXTRA_DATA, $colName);
             } else {
-                $sortingColumn = $ruleParts[1];
+                $colName = sprintf('t.%s -> \'%s\'', self::COLUMN_EXTRA_DATA, $colName);
             }
-
-            $result[] = isset($ruleParts[3]) && $ruleParts[3] === 'numeric'
-                ? sprintf("cast(nullif(%s, '') as decimal) %s", $sortingColumn, $ruleParts[2])
-                : sprintf('%s %s', $sortingColumn, isset($ruleParts[2]) ? $ruleParts[2] : 'ASC');
+            $this->queryParams[] = $colName;
+            $sortingColumn = '?';
```

Comment:
This doesn't work in MySQL, and probably in Postgresql neither. No error, but no effect. Since when `?` query parameter is filled in by PDO, then parameter is assumed as string and replaced as `ORDER BY 'JSON_EXTRACT(t.extra_data, \'$.fieldname\')' desc`, wrapped in quotes. If you wouldn't use query parameter, `?`, it would work fine 👍🏻

Reply:
nice catch, thank you. probably i test it only in query console